app/models/floor.rb

```diff
@@ -0,0 +1,9 @@
+# The model representing a floor associated with the HPI
+class Floor < ApplicationRecord
+  has_many :rooms, dependent: :nullify
+  belongs_to :building, optional: true # optional, because floors should be creatable without creating a building
```

Comment:
```suggestion
  belongs_to :building
```
I would remove optional, there cannot be a floor outside of a building (just assure the building is created before the floor)